common/src/test/java/org/conscrypt/java/security/cert/X509CertificateTest.java

```diff
@@ -318,6 +323,39 @@ private static X509Certificate certificateFromPEM(Provider p, String pem)
                 new ByteArrayInputStream(pem.getBytes(Charset.forName("US-ASCII"))));
     }
 
+    private static List<Pair<Integer, String>> normalizeGeneralNames(Collection<List<?>> names) {
+        // Extract a more convenient type than Java's Collection<List<?>>.
+        List<Pair<Integer, String>> result = new ArrayList<Pair<Integer, String>>();
+        for (List<?> tuple : names) {
+            assertEquals(2, tuple.size());
+            int type = ((Integer) tuple.get(0)).intValue();
+            // TODO(davidben): Most name types are expected to have a String value, but some use
+            // byte[]. Update this logic when testing those name types. See
+            // X509Certificate.getSubjectAlternativeNames().
+            String value = (String) tuple.get(1);
+            result.add(Pair.of(type, value));
+        }
+        // Although there is a natural order (the order in the certificate), Java's API returns a
+        // Collection, so there is no guarantee of the provider using a particular order. Normalize
+        // the order before comparing.
+        result.sort(new Comparator<Pair<Integer, String>>() {
```

Comment:
Sorry, but `List.sort()` is a JDK 1.8 method and we still test on 1.7 which unfortunately means the unit tests also have to be 1.7-compatible (we only just dropped support for Honeycomb and Java 1.6 last year!).  You should be able to use `Collections.sort()`.

(in an ideal world this would use a Guava MultiMap and Truth's assertThat for testing)

Reply:
Done. (I didn't even know this was a new method.)